settings.php

```diff
@@ -25,8 +25,19 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$ADMIN->add('reports', new admin_externalpage('reportallbackups', get_string('pluginname', 'report_allbackups'),
-    "$CFG->wwwroot/report/allbackups/index.php", 'report/allbackups:view'));
+if ($CFG->report_allbackups_isactive) {
+    $ADMIN->add('reports', new admin_externalpage(
+        'reportallbackups',
+        get_string('pluginname', 'report_allbackups'),
+        "$CFG->wwwroot/report/allbackups/index.php",
+        'report/allbackups:view'
+    ));
+}
 
-// No report settings.
-$settings = null;
+$settings = new admin_settingpage('report_allbackups', 'AllBackups');
+$settings->add(new admin_setting_configcheckbox(
+    'report_allbackups_isactive',
```

Comment:
This is the incorrect name for a plugin setting as mentioned above - plugins must store their settings in teh config_plugins table so the correct way to do this is to prefix it with the plugin name and slash - eg:
'report_allbackups/isactive'

but "isactive" is also incorrect - it would be better to use a single word like "visible" or "enabled" - personally I prefer "enabled" because this is similar to other plugin usage in Moodle.  (naming stuff is always hard.) :-)
Please convert this to "report_allbackups/enabled" and tidy up the asssociated  lang strings.